### PR TITLE
Chore: Increased test coverage for programming exercises (participations and submissions)

### PR DESCRIPTION
**Pull request by ungaralex**
### Description
Increases test coverage for the following resources:
- `PlantUmlResource`
- `ProgrammingExerciseParticipationResource`
- `ProgrammingSubmissionResource`

Also adds new request mocks for Bamboo requests and a new DTO, which partially solves #1024.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/connectors/BambooService.java [hunk 1]
```diff
@@ -732,47 +735,37 @@ private Long calculateScoreForResult(Result result, int skippedTests) {
      * - buildTestSummary:      a string generated by Bamboo summarizing the build result
      * - buildCompletedDate:    the completion date of the build
      */
+    // TODO Create a DTO for the returned result
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/BambooService.java]: please fix the TODO

File: src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/dto/BambooBuildResultDTO.java [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+
+    private String buildReason;
+
+    private ZonedDateTime buildCompletedDate;
+
+    private BambooTestResultsDTO testResults;
+
+    private String vcsRevisionKey;
+
+    // TODO add proper DTOs for the following two attributes
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/dto/BambooBuildResultDTO.java]: please fix the TODO

File: src/main/java/de/tum/in/www1/artemis/service/connectors/BambooService.java (hunk 1 above)
Comment (ungaralex) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/BambooService.java]: Done

File: src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/dto/BambooBuildResultDTO.java (hunk 2 above)
Comment (ungaralex) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/dto/BambooBuildResultDTO.java]: Done

**Comment by ungaralex** [comment]
I added the missing DTOs, everything should be resolved now.

**Pull request merged**